addons/mass_mailing/tests/common.py

```diff
@@ -99,10 +99,21 @@ def assertMailTraces(self, recipients_info, mailing, records,
             recipient_trace = traces.filtered(
                 lambda t: t.email == email and t.state == state and (t.res_id == record.id if record else True)
             )
-            self.assertTrue(
-                len(recipient_trace) == 1,
-                'MailTrace: email %s (recipient %s, state: %s, record: %s): found %s records (1 expected)' % (email, partner, state, record, len(recipient_trace))
-            )
+            # try to find a "state" issue
+            if not recipient_trace:
+                state_trace = traces.filtered(
+                    lambda t: t.email == email and (t.res_id == record.id if record else True)
+                )
+                if state_trace:
+                    self.assertTrue(
+                        len(recipient_trace) == 1,
+                        'MailTrace: email %s (recipient %s, state: %s, record: %s): not found, but found trace with state %s' % (email, partner, state, record, state_trace.state)
```

Comment:
Hmmm.
That whole part is very strange.
How can "len(recipient_trace)" be equal to one if it's Falsy here above?